### ISSUE TITLE
Add link to Settings page in plugin action links

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#1194]
## Feature Description

Many plugins add a link to their settings page to the plugin row itself for convenience. 

![image](https://user-images.githubusercontent.com/1621608/75675907-cfd35100-5c90-11ea-9128-b27d91d6e1c0.png)

We should do the same to improve the discoverability of plugin settings and generally feel more consistent with other well-known plugins.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* In the WordPress plugins list table, the Site Kit plugin should have a "Settings" link (to the `googlesitekit-settings` admin screen) as first item in the links below (e.g. before "Deactivate") _if_ the current user has access to the screen (`Permissions::MANAGE_OPTIONS`).

## Implementation Brief

* Add a new class similar to `\Google\Site_Kit\Core\Admin\Plugin_Row_Meta` in the same folder called `Plugin_Action_Links`
* Add a `register` method with a `plugin_action_links` filter to the new class where the name of the filter is generated like so:
  `'plugin_action_links_' . GOOGLESITEKIT_PLUGIN_BASENAME`
* Add a closure function to the filter which:
  * Receives `$links` array as a parameter
  * Generates  a new settings url using `Context::admin_url` 
  * Adds the url to a link with 'Settings' text which is translatable
  * Adds the new settings link to the `$links` array using `array_unshift` so that the settings link is returned before the deactivate link
  * Returns the `$links` array
  * Add a conditional to check if the user has the ability to view the settings page (`Permissions::MANAGE_OPTIONS`) before the link is created and returned
* Call the method in the same way as `Plugin_Row_Meta` in the `register` method of `Plugin.php` 

## QA Brief

* The Settings Link will be shown on plugin once the plugin activated and initial setup is complete.  

## Changelog entry

* Add link to Settings page in plugin action links on plugins page.

**Comment by felixarntz** [comment]
@aaemnnosttv Potentially a link to the "Dashboard" (or splash if not connected) would be more suitable here? While most plugins have a "Settings" link I'd argue this is largely because for these plugins their "Settings" is the central entry point to the overall functionality. For us however, that is the "Dashboard".

I'd say this is mostly for discoverability, rather than providing a link to the plugin's settings. Also note that if we linked to "Settings" it would only appear sometimes because the menu is only available if you're an admin and already connected.

What do you think?

**Comment by aaemnnosttv** [comment]
@Hazlitte your IB describes adding a link to the Site Kit dashboard, but the issue is about linking to the settings page.

> * Edit `includes/Plugin.php`
> * Add a `plugin_action_links` filter to the `register` method where the name of the filter is generated like so:

Let's not change the main `Plugin` class more than we need to for this addition. The changes here are very similar to our filtering of `plugin_row_meta` for which we have a `\Google\Site_Kit\Core\Admin\Plugin_Row_Meta` class. It would make sense to use a similar approach for this as well.

> where the name of the filter is generated like so:
> `plugin_action_links_' . dirname(plugin_basename( __DIR__ )) . '/google-site-kit.php`

This looks like it would result in the correct filter name, however we should use Site Kit constants to construct this rather than hardcoding them. In this case `GOOGLESITEKIT_PLUGIN_BASENAME` should be the proper value to use for the `$plugin_file` portion of the filter name.  

> Generates  a new dashboard url using `add_query_arg`

We should use `Context::admin_url` for getting the necessary URL for our settings screen – no need to use `add_query_arg` directly. ([example](https://github.com/google/site-kit-wp/blob/db4d6444f12e4185f1ff36e2602e3b6fe14a1fc6/includes/Core/Admin/Screens.php#L263))

Finally, the IB is missing the conditional check in the ACs which only adds the link if the user has the ability to view the settings page.

**Comment by Hazlitte** [comment]
Thank you @aaemnnosttv, I have updated the IB accordingly.  The comment from @felixarntz indicates that it might be more suitable to add a dashboard link as opposed to a settings one.  Perhaps we could add both?

**Comment by felixarntz** [comment]
@Hazlitte This was a rather old comment. We revisited this recently and decided to add a Settings link only which is the common approach between many WordPress plugins.

**Comment by aaemnnosttv** [comment]
> Thank you @aaemnnosttv, I have updated the IB accordingly. The comment from @felixarntz indicates that it might be more suitable to add a dashboard link as opposed to a settings one. Perhaps we could add both?

@Hazlitte I figured that's probably what you were going on, but IBs should always be written against the ACs. Sometimes ACs need updating in the process, but the AC should _always_ be the source of truth.

---

The IB is on the right track now, just a few things to adjust:

>   `'plugin_action_links_' . dirname( GOOGLESITEKIT_PLUGIN_BASENAME ) . '/google-site-kit.php'`

This is still hardcoding pieces which are not necessary and already covered by the `GOOGLESITEKIT_PLUGIN_BASENAME` constant:

```
>>> dirname( GOOGLESITEKIT_PLUGIN_BASENAME ) . '/google-site-kit.php'
=> "google-site-kit/google-site-kit.php"
>>> GOOGLESITEKIT_PLUGIN_BASENAME
=> "google-site-kit/google-site-kit.php"
```

That is, `'plugin_action_links_' . GOOGLESITEKIT_PLUGIN_BASENAME` is all you need here 😄 

> * Add the new class to the `autolooad_classmap`

This step is unnecessary as the classmap is generated by Composer and not under version control.

**Comment by Hazlitte** [comment]
Many thanks @aaemnnosttv, I have updated the IB accordingly.

**Comment by aaemnnosttv** [comment]
IB ✅
